ios-target/View/Home/TopicsViewController.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  TopicsViewController.swift
+//  ios-target
+//
+//  Created by sol manini on 7/1/19.
+//  Copyright © 2019 TopTier labs. All rights reserved.
+//
+
+import UIKit
+
+class TopicsViewController: ModalViewController {
+  
+  @IBOutlet weak var topicsCollectionHeightContraint: NSLayoutConstraint!
+  @IBOutlet weak var topicsCollectionView: UICollectionView!
+  
+  var viewModel = TopicsViewModel()
+  
+  weak var delegate: TopicViewModelDelegate?
```

Comment:
I'm guessing this view controller delegate won't implement `didUpdateState` and only `topicSelected`. In that case, it might be helpful to provide a default implementation of `didUpdateState` in a `TopicViewModelDelegate ` protocol extension and thus making it optional for clients.

Reply:
yep, used to load topics on the topicViewModel but then realized i needed them on the homevieModel too so I changed the logic and forgot about this unused state update!